### ISSUE TITLE
Generated class does not have import statement for generic type in different package

### ISSUE THREAD

**Issue by nebhale** [immutables/immutables#338]
Given the following three classes in three different packages:

```
package immutables;

public abstract class ParentWithGeneric<T> {
    public abstract T getItem();
}
```

```
package immutables.bravo;

import immutables.ParentWithGeneric;
import immutables.alpha.ImmutableSpecificType;
import org.immutables.value.Value;

@Value.Immutable
public abstract class ChildWithGeneric extends ParentWithGeneric<ImmutableSpecificType> {
}
```

```
package immutables.alpha;

import org.immutables.value.Value;

@Value.Immutable
public abstract class SpecificType {
}
```

The following class is generated for `ImmutableChildWithGeneric`.

```
package immutables.bravo;

import immutables.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import javax.annotation.Generated;

/**
 * Immutable implementation of {@link ChildWithGeneric}.
 * <p>
 * Use the builder to create immutable instances:
 * {@code ImmutableChildWithGeneric.builder()}.
 */
@SuppressWarnings("all")
@Generated({"Immutables.generator", "ChildWithGeneric"})
public final class ImmutableChildWithGeneric extends ChildWithGeneric {
  private final ImmutableSpecificType item;
...
```

This code, generated by 2.2 references the `ImmutableSpecificType` class, but it does not include an `import` statement that would reach it at `immutable.alpha.ImmutableSpecificType`.  This results in the inability to compile the `ImmutableChildWithGeneric` type.

**Comment by elucash** [comment]
wow, cool problem, thank you for revealing that! This definitely have to be fixed

**Comment by nebhale** [comment]
Just double checked the `SNAPSHOT` of this and it looks good.  Thanks for the quick work.

**Comment by elucash** [comment]
Released in 2.2.1 Thank you!
